associations.rb

```diff
@@ -46,26 +50,38 @@ def generate_migrations
 
 
 class Hotel < ActiveRecord::Base
-  #insert our associations here
+  has_many :rooms
+  has_many :bookings, :through => :rooms
  
   def to_s
     "#{name} with #{rooms.count} rooms"
   end
+
+  def booked_guests
+    [].tap do |r|
+      self.bookings.each { |b| r << b.guest }
+    end
```

Comment:
This is simpler code: `self.bookings.map { |booking| booking.guest }`.
But more importantly, this is going to generate a separate query for each guest (an "n+1" problem).